### PR TITLE
Wizard recipe: Chemfiles-v0.9.2

### PR DESCRIPTION
**Pull request by Luthaf**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: Chemfiles
* Version: v0.9.2

@staticfloat please review and merge.

### CODE REVIEW

**Comment by Luthaf** [comment]
As someone who went through a bunch of different ways to build C++ code for use in Julia (WinRPM on openSuse build, conda recipes, direct download), the new BinaryBuilder/Yggdrasil workflow is way better than previous solutions!

Congrats for this :tada: !!

My main pain point was that documentation was hard to find, and scattered around multiple places.

**Comment by giordano** [comment]
> My main pain point was that documentation was hard to find, and scattered around multiple places.

The main documentation is at https://juliapackaging.github.io/BinaryBuilder.jl/dev/, didn't you find this?  In addition, we recently started to collect common [build tricks](https://github.com/JuliaPackaging/Yggdrasil/wiki/Build-Tricks) in the wiki to help user tackle common issues.

**Comment by Luthaf** [comment]
I found it, but I was using julia 1.0.5, and I could not get it to work. Switching to julia 1.3.1, I ran into https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/593, and I got it to work with 1.3.0.

Since I don't follow julia development very closely, I had missed the fact that 1.3 had a new way to load binary artifacts

File: C/Chemfiles/bundled/patches/fileno_posix.patch [hunk 1]
```diff
@@ -0,0 +1,24 @@
+--- a/external/netcdf/libdispatch/dfile.c
++++ b/external/netcdf/libdispatch/dfile.c
+@@ -32,7 +32,9 @@
+ #ifdef HAVE_STDIO_H
+ #include <stdio.h>
+ #endif
++#ifdef  __USE_POSIX
```
Comment (Luthaf) [inline C/Chemfiles/bundled/patches/fileno_posix.patch]: thanks for taking the time to do this! Can I incorporate this patch directly in our fork of netcdf? https://github.com/chemfiles/netcdf-c

Comment (giordano) [reply C/Chemfiles/bundled/patches/fileno_posix.patch]: Sure!

**Pull request merged**